### DISCUSSION TITLE
Selecting a testing framework

### DISCUSSION THREAD

**Discussion by liberty-rising** [idom-team/idom#704] [Question] [answered]
What's the recommended testing framework to use with IDOM? Looking to implement it in timeflow.

**Comment by rmorshea** [comment, marked as answer]
In short, I think the recommendation is to use an async web driver like [playwrite](https://github.com/microsoft/playwright), [`arsenic`](https://arsenic.readthedocs.io/en/latest/reference/session.html), or [`pyppeteer`](https://miyakogi.github.io/pyppeteer/reference.html) to run UI automation tests. I'll and put together a code sample at some point that uses one of these tools, or if you need something sooner, feel free to book a slot during [office hours](https://github.com/idom-team/idom/discussions/312).

IDOM does provide some [testing utilities](https://idom-docs.herokuapp.com/docs/_auto/apis.html#module-idom.testing) of it's own, but this relies on the sync Selenium web driver. I plan on changing this [soon though](https://github.com/idom-team/idom/pull/703) because synchronous web drivers present [some problems](https://github.com/idom-team/idom/issues/591). I'm not really sure when this work will be done though as it will probably be quite sweeping in its impact.

**Reply by liberty-rising** [reply]
Sounds good, thanks for the quick answer. I'll try them out and see if I can get something working.

**Reply by rmorshea** [reply]
@liberty-rising, I just found [playwrite](https://github.com/microsoft/playwright) which seems to be much better maintained than either arsenic of pyppeteer. Assuming it works I think this is probably what IDOM's test suite will use.

**Reply by liberty-rising** [reply]
@rmorshea ahh it looks much better. Thanks for letting me know, I'll see when I could get the time to migrate our testing to playwrite in that case. Cheers!

**Reply by acivitillo** [reply]
only issue I see with playwrite is that tests are in js, while we are building apps in python

**Reply by acivitillo** [reply]
[scrap that](https://playwright.dev/python/docs/intro)
